x/treasury/internal/keeper/indicator.go

```diff
@@ -11,104 +12,89 @@ import (
+	alignedAmt = sdk.ZeroDec()
+	for _, coinReward := range coins {
+		if coinReward.Denom != denom {
+			swappedReward, err := k.marketKeeper.ComputeInternalSwap(ctx, coinReward, denom)
 			if err != nil {
 				continue
 			}
-			taxRewardInMicroSDR = taxRewardInMicroSDR.Add(swappedReward.Amount)
+			alignedAmt = alignedAmt.Add(swappedReward.Amount)
 		} else {
-			taxRewardInMicroSDR = taxRewardInMicroSDR.Add(coinReward.Amount)
+			alignedAmt = alignedAmt.Add(coinReward.Amount)
 		}
 	}
 
-	return taxRewardInMicroSDR
+	return alignedAmt
 }
 
-// SeigniorageRewardsForEpoch returns seigniorage rewards for the epoch
-func SeigniorageRewardsForEpoch(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	seignioragePool := k.PeekEpochSeigniorage(ctx, epoch)
-	rewardAmt := k.GetRewardWeight(ctx, epoch).MulInt(seignioragePool)
-	seigniorageReward := sdk.NewDecCoinFromDec(core.MicroLunaDenom, rewardAmt)
+// UpdateIndicators updates interal indicators
+func (k Keeper) UpdateIndicators(ctx sdk.Context) {
+	epoch := core.GetEpoch(ctx)
+	lunaTotalBondedAmount := k.stakingKeeper.TotalBondedTokens(ctx)
 
-	microSDRReward, err := k.marketKeeper.ComputeInternalSwap(ctx, seigniorageReward, core.MicroSDRDenom)
-	if err != nil {
-		return sdk.ZeroDec()
-	}
+	// Compute Tax Rewards & Tax Rewards per (staked)Luna
+	taxRewards := sdk.NewDecCoins(k.PeekTaxProceeds(ctx))
+	TR := k.alignCoins(ctx, taxRewards, core.MicroSDRDenom)
+	TRL := TR.QuoInt(lunaTotalBondedAmount)
 
-	return microSDRReward.Amount
-}
+	k.SetTRL(ctx, epoch, TRL)
 
-// MiningRewardForEpoch returns the sum of tax and seigniorage rewards for the epoch
-func MiningRewardForEpoch(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	taxRewards := TaxRewardsForEpoch(ctx, k, epoch)
-	seigniorageRewards := SeigniorageRewardsForEpoch(ctx, k, epoch)
+	// Reset tax proceeds after computing TRL for the next epoch
+	k.SetTaxProceeds(ctx, sdk.Coins{})
 
-	return taxRewards.Add(seigniorageRewards)
-}
+	// Compute Seigniorage Rewards
+	seigniorage := k.PeekEpochSeigniorage(ctx)
+	seigniorageRewardsAmt := k.GetRewardWeight(ctx).MulInt(seigniorage)
+	seigniorageRewards := sdk.DecCoins{sdk.NewDecCoinFromDec(core.MicroLunaDenom, seigniorageRewardsAmt)}
+	SR := k.alignCoins(ctx, seigniorageRewards, core.MicroSDRDenom)
 
-// TRL returns tax rewards per luna at the epoch
-func TRL(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	return UnitLunaIndicator(ctx, k, epoch, TaxRewardsForEpoch)
-}
+	k.SetSR(ctx, epoch, SR)
 
-// SRL returns Seigniorage rewards per luna at the epoch
-func SRL(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	return UnitLunaIndicator(ctx, k, epoch, SeigniorageRewardsForEpoch)
-}
+	// Compute Mining Rewards
+	MR := TR.Add(SR)
 
-// MRL returns mining rewards per luna at the epoch
-func MRL(ctx sdk.Context, k Keeper, epoch int64) sdk.Dec {
-	return UnitLunaIndicator(ctx, k, epoch, MiningRewardForEpoch)
+	k.SetMR(ctx, epoch, MR)
```

Comment:
Do we need to store MR as a separate variable in the keeper? it seems to be a straight up additive result of seigniorage rewards and tax rewards